src/main/java/tetris/Controller.java

```diff
@@ -32,24 +32,31 @@
         Platform.runLater(() -> Action.SOFT_DROP.attempt(tetromino, grid));
         Platform.runLater(() -> redraw());
     });
-
     private Settings settings;
+    private TetrominoFactory factory;
 
     /**
      * the Controller class determines the game flow and does the actual event
      * handling.
-     * 
+     *
      * @param ui
-     *            the application in which the game is running
+     * @param settings
+     * @param factory
+     * @param soundManager
+     * @param score
+     * @param scoreBoard
      */
-    public Controller(View ui, Settings settings) {
+    public Controller(View ui, Settings settings, TetrominoFactory factory,
```

Comment:
What would you suggest to do? I prefer to pass this as a parameter so another Factory can be inserted. A Factory that, for example, generates other pieces. Besides that, I don't see how we can prevent voilation of Dependecy Inversion without doing it this way.